c7n/resources/sagemaker.py

```diff
@@ -66,14 +68,107 @@ class resource_type(object):
         service = 'sagemaker'
         enum_spec = ('list_training_jobs', 'TrainingJobSummaries', None)
         detail_spec = (
-            'describe_training_job', 'TrainingJobName',
-            'TrainingJobName', None)
+            'describe_training_job', 'TrainingJobName', 'TrainingJobName', None)
         id = 'TrainingJobArn'
         name = 'TrainingJobName'
         date = 'CreationTime'
         dimension = None
         filter_name = None
 
+    permissions = (
+        'sagemaker:ListTrainingJobs', 'sagemaker:DescribeTrainingJobs',
+        'sagemaker:ListTags')
+
+    def __init__(self, ctx, data):
+        super(SagemakerJob, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(
+            self.data.get('query', [
+                {'StatusEquals': 'InProgress'}]))
+
+    def resources(self, query=None):
+        q = self.query_filter()
+        if q is not None:
+            query = query or {}
+            for i in range(0, len(q)):
+                query[q[i]['Name']] = q[i]['Value']
+        return super(SagemakerJob, self).resources(query=query)
+
+    def query_filter(self):
```

Comment:
this method feels like it would be better on the query parser class, ie having the parse output being the actual query fit for use by the resource manager, and encoding the default as a class/instance variable.